content/powershell/concepts/objects/objects.md

```diff
@@ -1,13 +1,147 @@
 Subjects:
+  - 'Code Foundations'
   - 'Computer Science'
+  - 'Bash/Shell'
 Tags:
-  - 'Objects'
+  - 'Classes'
   - 'Data Types'
   - 'Data Structures'
+  - 'Hash Maps'
+  - 'Methods'
+  - 'Objects'
+  - 'OOP'
+  - 'Properties'
 CatalogContent:
   - 'learn-powershell'
   - 'paths/computer-science'
 ---
+
+Everything in PowerShell is an **object**. An object is a combination of variables and functions. Each object has:
+
+- **Properties**: variables that describe the object (characteristics)
+- **Methods**: functions that describe how to use the object (actions)
+
+Each object is an instance of a blueprint called a *type*, or *[class](https://en.wikipedia.org/wiki/Class_(computer_programming))*. 
+
+## Getting Properties and Methods of an Object
+
+All information associated with an object is called a member. The `Get-Member` cmdlet reveals the member of an object – properties, methods, and so on. It also shows the type of an object.
+
+```shell
+"Codecademy" | Get-Member
+```
+
+## Object Properties
+
+Properties, or attributes, are characteristics of an object. String objects, for example, have a property called `Length` which holds the number of characters in a string.
+
+```shell
+PS > "hello".Length # "hello" has 5 characters
+5
+```
+
+The `Get-Member` cmdlet can be utilized with the `MemberType` parameter to only show the properties of an object. Consider a script file:
+
+```shell
+PS > Get-ChildItem .\script.ps1 | Get-Member -MemberType Property # Shows all properties of the script
+   
+  TypeName: System.IO.FileInfo
+
+Name              MemberType Definition
+----              ---------- ----------
+Attributes        Property   System.IO.FileAttributes Attributes {get;set;}
+CreationTime      Property   datetime CreationTime {get;set;}
+CreationTimeUtc   Property   datetime CreationTimeUtc {get;set;}
+Directory         Property   System.IO.DirectoryInfo Directory {get;}
+DirectoryName     Property   string DirectoryName {get;}
+Exists            Property   bool Exists {get;}
+Extension         Property   string Extension {get;}
+FullName          Property   string FullName {get;}
+IsReadOnly        Property   bool IsReadOnly {get;set;}
+LastAccessTime    Property   datetime LastAccessTime {get;set;}
+LastAccessTimeUtc Property   datetime LastAccessTimeUtc {get;set;}
+LastWriteTime     Property   datetime LastWriteTime {get;set;}
+LastWriteTimeUtc  Property   datetime LastWriteTimeUtc {get;set;}
+Length            Property   long Length {get;}
+LinkTarget        Property   string LinkTarget {get;}
+Name              Property   string Name {get;}
+
+PS > (Get-ChildItem .\script.ps1).CreationTime # Prints the date when the script was created
+Thursday, January 19, 2023 3:34:43 AM
+```
+
+## Object Methods
+
+Methods are the actions an object can take. One of the methods of a String object, for example, is `Contains()`. This method returns `True` if a string contains the specified substring, otherwise `False`.
```

Comment:
```suggestion
Methods are the actions an object can take. One of the methods of a `String` object, for example, is `Contains()`. This method returns `True` if a string contains the specified substring, otherwise `False`.
```

Reply:
or lowercase "string"